doc/_index.md

```diff
@@ -1,13 +1,85 @@
+database abstraction and a publisher / subscriber interface.  Another important
+concept is the fact that users own the data they send to the API.  Let's have a
+quick look at how this all fits together.
+
+### Database Abstraction
+
+All the data managed by KernelCI is stored in a MongoDB database using node
+objects.  These can contain data about any part of the testing hierarchy such
+as a kernel revision, a build, static test results, runtime functional tests,
+regressions etc.  Each node has a parent so they form a tree
+([DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph)).  There is
+typically one root node for each kernel revision with lots of child nodes
+containing all the test data that relates to it.
+
+Each node object also has a state which can be used when orchestrating the
+pipeline.  For example, a node will be in the Running state while awaiting some
+results.  There's also a result value, to tell whether the job passed or
+failed.  Finally there's a list of artifacts with URLs to know where to find
+all the related files (binaries, logs, generated results etc.).
+
+Here's a publicly available
+[graph](https://charts.mongodb.com/charts-kernelci-api-staging-otxuy/public/dashboards/5b52c3c6-81bb-4658-a5f3-8ebd2e980436#)
+showing the number of nodes added to the database every day:
+
+<div style="padding: 0 0 20px 0"><iframe style="background: #FFFFFF;border: none;border-radius: 2px;box-shadow: 0 2px 10px 0 rgba(70, 76, 79, .2);" width="640" height="480" src="https://charts.mongodb.com/charts-kernelci-api-staging-otxuy/embed/charts?id=64c9fa16-ba81-40e9-8cd7-cd56fb88fbf3&maxDataAge=3600&theme=light&autoRefresh=true"></iframe></div>
+
+> **Note:** The API doesn't manage storage, the only requirement is to provide
+> publicly-available HTTP(s) URLs for each artifact.
+
+### Publisher / Subscriber Interface
+
+Every time some data changes in the database, basically every time a node is
+has been added or updated, an event is sent on the Publisher / Subscriber
+interface (Pub/Sub).  For example, when a new kernel revision is found and a
+new node is created for it, an event will be sent to tell subscribers about it
+with something like "A checkout node has been created for kernel revision
+v6.2-rc4".  The actual event is a [CloudEvents](https://cloudevents.io) object
+with some JSON data containing a subset of the corresponding node database
+entry.
+
+Any client code can subscribe to receive events with an API token and implement
+features based on how to handle these events.  The API generates events
+automatically whenever nodes are changing but clients may also use the
+interface to publish their own events and coordinate other parts of the
+pipeline too.
+
+### User Ownership
+
+Interacting with the API requires a token which is associated with a particular
+user.  Whenever a user sends some data such as a new node, it is owned by that
+user.  While all nodes are publicly readable, only the owner of the node can
```

Comment:
I suppose admin has superuser rights which overrides this rule - should it be stated explicitly or is it irrelevant in this context?

Reply:
I don't think that's relevant here as this is an overview - and in fact there appears to be a bug so even a user in the "admin" group can't update nodes apparently, see https://github.com/kernelci/kernelci-pipeline/issues/315